File: testdata/fixtures/ruby/vulnerable/log_injection.rb
# Vulnerable: Log injection with unsanitized user input
# Expected: GTSS-LOG-001 (UnsanitizedLogInput)

class PaymentsController < ApplicationController
  def create
    amount = params[:amount]
    Rails.logger.info "Payment initiated: amount=#{params[:amount]} user=#{params[:user_id]}"

    begin
      process_payment(amount)
      Rails.logger.info "Payment successful for params[#{params[:transaction_id]}]"
    rescue => e
      Rails.logger.error "Payment failed: #{params[:error_detail]} - #{e.message}"
    end
  end

  def refund
    Rails.logger.warn "Refund requested by user: #{params[:user_id]} for order #{params[:order_id]}"
    process_refund(params[:order_id])
  end

  def webhook
    Rails.logger.debug "Webhook received: #{request.raw_post}"
    process_webhook(request.body.read)
  end
end

File: testdata/fixtures/python/vulnerable/log_injection.py
# Vulnerable: Log injection with unsanitized user input
# Expected: BATOU-LOG-001 (UnsanitizedLogInput)

import logging
from flask import Flask, request

app = Flask(__name__)
logger = logging.getLogger(__name__)


@app.route("/login", methods=["POST"])
def login():
    username = request.form.get("username")
    logger.info(f"Login attempt for user: {request.form['username']}")

    if authenticate(username, request.form.get("password")):
        logger.info(f"Login successful: {request.form['username']}")
        return "OK"
    else:
        logger.warning(f"Failed login for: {request.form['username']}")
        return "Unauthorized", 401


@app.route("/api/data", methods=["POST"])
def process():
    logging.info(f"Processing request from: {request.form['user_id']}")
    data = request.json
    logging.debug(f"Request payload: {request.data}")
    return "OK"


@app.route("/search")
def search():
    query = request.args.get("q")
    logger.info(f"Search query: {request.args['q']}")
    return "Results"


def authenticate(username, password):
    return True
